pkgs/misc/cups/drivers/cnijfilter2/default.nix

```diff
@@ -4,11 +4,11 @@
 stdenv.mkDerivation {
   pname = "cnijfilter2";
 
-  version = "5.70";
+  version = "5.90";
 
   src = fetchzip {
-    url = "http://gdlp01.c-wss.com/gds/0/0100009930/01/cnijfilter2-source-5.70-1.tar.gz";
-    sha256 = "045zjsmaidn1m44ki6m1018gjzbj77gm234n5i2lshxpbzpyh0is";
+    url = "http://gdlp01.c-wss.com/gds/4/0100010484/01/cnijfilter2-source-5.90-1.tar.gz";
```

Comment:
How does one discover these releases? The latest I can find on the Canon website is 5.20 or 5.30

Reply:
Hi Pavol, 

I'm sure there are beter ways but here is my story:

1. bought new printer (Canon G5040),
2. found out my current drivers for CUPS server don't support my new printer,
3. searched for canon drivers in NixOS package repo with no luck (drivers description does not have printer list so it's not indexed),
4. browsed to https://www.canon-europe.com/support/consumer_products/products/fax__multifunctionals/inkjet/pixma_g_series/pixma-g5040.html?type=drivers&language=en&os=linux%20(64-bit) to figure out linux drivers,
5. went back to NixOS package search and queried for cnijfilter phrase,
6. found older version of the drivers in cnijfilter2 package,
7. tested older version with no luck,
8. recompiled the package with new version url- tada :+1:,
9. decided to share back my happiness with the community.

Happy end of my story.

I think you could generalize that like that:

1. figure out the newest Canon printer model,
2. browse to support page,
3. compare drivers version from support page with the one available at nixpkgs.

Reply:
Yeah, I was afraid it was almost impossible to tell which is the latest version if you don't know what's the newest Canon printer model, but hoped for a better way. Thank you for the explanation!